components/hab/Cargo.toml

```diff
@@ -38,6 +38,9 @@ path = "../core"
 [dependencies.habitat_common]
 path = "../common"
 
+[dependencies.habitat_sup]
```

Comment:
Do we want `hab` the CLI to have a dependency on the supervisor?  That doesn't sit right with me, from an architectural standpoint.  It feels like it would be better to have both `hab` and `sup` depend on the same auxiliary crate.

Reply:
Since `hab` actually delegates some commands to `sup`, I would say that the dependency is logical, isn't it?

Reply:
One question I have is how much does the size of the hab cli increase if we pull in an explicit supervisor dependency?

Reply:
It would definitely be possible to move the CLAP config that `hab` and `hap-sup` share to an independent crate, but there would still be an implicit `hab` → `hab-sup` dependency as long as there are commands like `hab svc start` that exec `hab-sup` from the `hab` binary. This doesn't prevent `hab-sup` changes from breaking the invocation from `hab`, but it at least keeps the usage notes in sync.

To me, the argument for keeping the CLAP config in the `hab-sup` crate rather than a third one that `hab` and `hab-sup` share is that the commands logically belong to `hab-sup` and are just aliased from `hab`. Furthermore, it's closer to the existing structure and makes for a smaller change to keep it this way.

As for the size of the `hab` executable, when I run `cargo build; ls -lh ../../target/debug/hab` I get
* 80M for `master`
* 109M for `jbauman/issues/3130`

Even running with `cargo build --release`, the sizes are much larger than the installed `/hab/pkgs/core/hab/0.38.0/20171026215954/bin/hab`, which is only ~9M. There must be additional stripping/dynamic linking that I'm not sure how to do.

Reply:
This might be splitting hairs, but I think there's a distinction between delegation and a dependency.  `hab` delegates certain commands to `hab-sup`. If `hab-sup` is present, it receives those commands, does its thing and everything's awesome.  If `hab-sup` is not present, there's an error, but `hab` itself remains fully functional. In my opinion, that's very different from saying that `hab-sup` is required in order to build `hab`. It feels a bit like saying `builder-api` is a dependency of `builder-api-client`. One is a client, one is a server. 

It's also worth noting that building the `hab` CLI via the `mac-build.sh` script fails on this branch (but works on `master`). I wonder if that's related to including `hab-sup` as a dependency.

Reply:
I think we see this the same way. I agree there's not an operational dependency on `hab-sup` for the execution of commands like `hab svc start`, but I _do_ think there's a real dependency for the CLI usage. If the one shared copy of the CLI docs is out of line with the implementation, there's not much to be done, but with making an actual code dependency from `hab` → `hab-sup`, it should help avoid `hab svc start --help` displaying stale docs when updated ones exist within the `sup` crate.

As for the `mac-build.sh` script. I can't seem to get it to work on master either, so I'm having trouble determining whether this change has broken it. It's either not a part of the automated tests, or doesn't break anything there. I'll ask in channel to try to figure out the issue.

Reply:
My 2c: the issue I see with taking a static dependency on the clap arguments, but keeping a dynamic dependency on hab-sup itself, is that we can still get out of sync.  So neither a shared crate, or a static dependency fully solves the aliasing issue.  Ideally we would have consistency in terms of a static vs. dynamic dependency architecturally. @reset or @fnichol may want to weigh in as well.